### DISCUSSION TITLE
1-1 mapping [Jul 21]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2871] [Discord Threads]
> 🙋‍♂️ **lublak**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### You can consider creating a table for `number` and `description`. Let's call it option table.   
####    
#### From your source table, create a link (LTAR) for option table. You can establish link between source table & option table by using this field. Add another look-up in source table to display associated number from option table.  
#  
> 🙋‍♂️ **lublak**  
#### <@857998597474484295> it has only "many" options:  
![image0](https://cdn.discordapp.com/attachments/999306492635262976/999310104828596354/unknown.png)  
#  
> 👨‍🦲 **dstala**  
#### There is no explicit 1-1 mapping. We can use `Single select` column type, but then, we will not be able to get associated index  
#  
> 🙋‍♂️ **lublak**  
#### Would this be a feature request?  
#  
> 👨‍🦲 **dstala**  
#### It can be a feature request. You can help us understand your use-case in detail when you request.  
#  
> 🙋‍♂️ **lublak**  
#### I have a table where numbers have to be entered. These number codes are stored in the database by another system with a description.   
#### ```   
#### code_description_table   
#### | number | description |   
#### | ------ | ----------- |   
#### | 890    | test        |   
#### | 840    | work        |   
#### ```   
#### Now I have a second table with inputs.   
#### ```   
#### input_table   
#### | somefield | anotherfield | code |   
#### | --------- | ------------ | ---- |   
#### | blabliblu | lala         | 890  |   
#### ```   
#### Of course, it is not pleasant to work with such numbers.   
#### Therefore, a dropdown would be really cool which displays the description in the dropdown instead of the number (i.e. code_description_table.description). So link record but only with a single value.   
#### With input_table.code, it would make sense to place a foreign key on code_description_table.number.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/999306492635262976)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
